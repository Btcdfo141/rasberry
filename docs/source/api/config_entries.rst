.. _config_entries_module:

:mod:`spencerassistant.config_entries`
-----------------------------------

.. automodule:: spencerassistant.config_entries
    :members:
