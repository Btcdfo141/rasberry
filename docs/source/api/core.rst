.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: