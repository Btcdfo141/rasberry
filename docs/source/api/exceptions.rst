.. _exceptions_module:

:mod:`spencerassistant.exceptions`
-------------------------------

.. automodule:: spencerassistant.exceptions
    :members:
