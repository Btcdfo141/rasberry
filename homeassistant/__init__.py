"""Init file for Home Assistant."""


test2
