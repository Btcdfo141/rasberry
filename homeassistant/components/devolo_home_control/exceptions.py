"""Custom exceptions for the devolo_home_control integration."""


class CredentialsInvalid(Exception):
    """Given credentials are invalid."""
