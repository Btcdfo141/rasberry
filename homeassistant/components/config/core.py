"""Component to interact with Hassbian tools."""
import asyncio

from homeassistant.components.http import HomeAssistantView
from homeassistant.config import async_check_ha_config_file


@asyncio.coroutine
def async_setup(hass):
    """Set up the Hassbian config."""
    hass.http.register_view(CheckConfigView)
    return True


class CheckConfigView(HomeAssistantView):
    """Hassbian packages endpoint."""

    url = '/api/config/core/check_config'
    name = 'api:config:core:check_config'

    @asyncio.coroutine
    def post(self, request):
        """Validate configuration and return results."""
        hass = request.app['hass']

        if hass.components.hassio.is_hassio():
            errors = yield from hass.components.hassio.async_check_config()
        else:
            errors = yield from async_check_ha_config_file(hass)

        state = 'invalid' if errors else 'valid'

        return self.json({
            "result": state,
            "errors": errors,
        })
