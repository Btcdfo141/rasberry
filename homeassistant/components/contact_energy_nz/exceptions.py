"""Custom Exceptions."""

from homeassistant.exceptions import HomeAssistantError


class CannotConnect(HomeAssistantError):
    """Error to indicate we cannot connect."""


class InvalidAuth(HomeAssistantError):
    """Error to indicate there is invalid auth."""


class CannotUpdate(HomeAssistantError):
    """Error to indicate there is invalid auth."""
