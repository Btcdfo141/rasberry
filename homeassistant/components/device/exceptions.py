"""Device exceptions."""
from homeassistant.exceptions import HomeAssistantError


class InvalidDevice(HomeAssistantError):
    """When device is invalid."""


class DeviceNotFound(HomeAssistantError):
    """When referenced device not found."""
