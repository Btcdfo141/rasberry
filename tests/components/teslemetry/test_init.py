"""Test the Tessie init."""

from tesla_fleet_api.exceptions import InvalidToken, PaymentRequired, TeslaFleetError

from homeassistant.config_entries import ConfigEntryState
from homeassistant.core import HomeAssistant

from . import setup_platform


async def test_load_unload(hass: HomeAssistant) -> None:
    """Test load and unload."""

    entry = await setup_platform(hass)
    assert entry.state is ConfigEntryState.LOADED
    assert await hass.config_entries.async_unload(entry.entry_id)
    await hass.async_block_till_done()
    assert entry.state is ConfigEntryState.NOT_LOADED


async def test_auth_failure(hass: HomeAssistant) -> None:
    """Test init with an authentication error."""

    entry = await setup_platform(hass, side_effect=InvalidToken)
    assert entry.state is ConfigEntryState.SETUP_ERROR


async def test_subscription_failure(hass: HomeAssistant) -> None:
    """Test init with an client response error."""

    entry = await setup_platform(hass, side_effect=PaymentRequired)
    assert entry.state is ConfigEntryState.SETUP_ERROR


async def test_other_failure(hass: HomeAssistant) -> None:
    """Test init with an client response error."""

    entry = await setup_platform(hass, side_effect=TeslaFleetError)
    assert entry.state is ConfigEntryState.SETUP_RETRY
