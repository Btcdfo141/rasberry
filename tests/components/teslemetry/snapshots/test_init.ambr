# serializer version: 1
# name: test_devices[1b5f91b943118e4b2261967ad7eb9167-device]
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'https://teslemetry.com/console',
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'teslemetry',
        'abd-123',
      ),
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'Tesla',
    'model': None,
    'name': 'Wall Connector',
    'name_by_user': None,
    'serial_number': '123',
    'suggested_area': None,
    'sw_version': None,
    'via_device_id': <ANY>,
  })
# ---
# name: test_devices[1fcdab2ebba80b8364fbfb06fd300e3e-device]
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'https://teslemetry.com/console',
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'teslemetry',
        'LRW3F7EK4KC700000',
      ),
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'Tesla',
    'model': 'Model 3',
    'name': 'Test',
    'name_by_user': None,
    'serial_number': 'LRW3F7EK4KC700000',
    'suggested_area': None,
    'sw_version': None,
    'via_device_id': None,
  })
# ---
# name: test_devices[433c34f54d8a4c90f9d5b202be1b7c8f-device]
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'https://teslemetry.com/console',
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'teslemetry',
        'LRW3F7EK4KC700000',
      ),
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'Tesla',
    'model': 'Model 3',
    'name': 'Test',
    'name_by_user': None,
    'serial_number': 'LRW3F7EK4KC700000',
    'suggested_area': None,
    'sw_version': None,
    'via_device_id': None,
  })
# ---
# name: test_devices[7e0f367246b94bb9eaaecd32b4909d5f-device]
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'https://teslemetry.com/console',
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'teslemetry',
        'bcd-234',
      ),
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'Tesla',
    'model': None,
    'name': 'Wall Connector',
    'name_by_user': None,
    'serial_number': '234',
    'suggested_area': None,
    'sw_version': None,
    'via_device_id': <ANY>,
  })
# ---
# name: test_devices[9c3ae015fc745c11adddd4da3e429427-device]
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'https://teslemetry.com/console',
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'teslemetry',
        'bcd-234',
      ),
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'Tesla',
    'model': None,
    'name': 'Wall Connector',
    'name_by_user': None,
    'serial_number': '234',
    'suggested_area': None,
    'sw_version': None,
    'via_device_id': <ANY>,
  })
# ---
# name: test_devices[a57fdbbf4549931033220d38a73358da-device]
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'https://teslemetry.com/console',
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'teslemetry',
        '123456',
      ),
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'Tesla',
    'model': 'Powerwall 2, Tesla Backup Gateway 2',
    'name': 'Energy Site',
    'name_by_user': None,
    'serial_number': None,
    'suggested_area': None,
    'sw_version': None,
    'via_device_id': None,
  })
# ---
# name: test_devices[f02c7e3eeb190f7ee413be87f2827f16-device]
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'https://teslemetry.com/console',
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'teslemetry',
        '123456',
      ),
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'Tesla',
    'model': 'Powerwall 2, Tesla Backup Gateway 2',
    'name': 'Energy Site',
    'name_by_user': None,
    'serial_number': None,
    'suggested_area': None,
    'sw_version': None,
    'via_device_id': None,
  })
# ---
# name: test_devices[f56a5ce7004d3eb028333882827720d0-device]
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'https://teslemetry.com/console',
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'teslemetry',
        'abd-123',
      ),
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'Tesla',
    'model': None,
    'name': 'Wall Connector',
    'name_by_user': None,
    'serial_number': '123',
    'suggested_area': None,
    'sw_version': None,
    'via_device_id': <ANY>,
  })
# ---
